sql/2023/01/lazyloaded-lcp-opportunity.sql

```diff
@@ -0,0 +1,66 @@
+# HTTP Archive query to get % of WordPress sites that lazy-load their LCP image.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/28
+CREATE TEMP FUNCTION
+  getLoadingAttr(attributes STRING)
+  RETURNS STRING
+  LANGUAGE js AS '''
+  try {
+    const data = JSON.parse(attributes);
+    const loadingAttr = data.find(attr => attr["name"] === "loading")
+    return loadingAttr.value
+  } catch (e) {
+    return "";
+  }
+''';
+
+WITH
+  lcp_stats AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    JSON_EXTRACT_SCALAR(payload, '$._performance.lcp_elem_stats.nodeName') AS nodeName,
+    JSON_EXTRACT_SCALAR(payload, '$._performance.lcp_elem_stats.url') AS elementUrl,
+    JSON_EXTRACT(payload, '$._performance.lcp_elem_stats.attributes') AS attributes,
+    JSON_EXTRACT(payload, '$._detected_apps.WordPress') AS wpVersion,
+    getLoadingAttr(JSON_EXTRACT(payload, '$._performance.lcp_elem_stats.attributes')) AS loading,
+  FROM
+    `httparchive.pages.2022_10_01_*`
```

Comment:
```suggestion
    `httparchive.pages.2022_11_01_*`
```

Should use November 2022 data as it now ready for use.

Reply:
@mukeshpanchal27 Fair point, but we don't necessarily have to do that. We are using the October data already in the analysis, so changing it wouldn't be good at this point.

FWIW all of the queries in this repo will have some specific month on them. That's okay, we shouldn't keep this up to date. Just using what's the latest when we write the query is a good approach.